### PR TITLE
malt: add v1.2.3, v1.2.4

### PR DESCRIPTION
**Pull request by svalat**
Register version `1.2.4` and passed `1.2.3` of `malt` package.

### CODE REVIEW

File: var/spack/repos/builtin/packages/malt/package.py [hunk 1]
```diff
@@ -15,12 +15,14 @@ class Malt(CMakePackage):
...
-    url = "https://github.com/memtt/malt/archive/v1.2.2.tar.gz"
+    url = "https://github.com/memtt/malt/releases/download/v1.2.3/malt-1.2.3.tar.bz2"
     maintainers("svalat")
 
     license("CECILL-C")
 
     # Versions
+    version("1.2.4", sha256="47068fe981b4cbbfe30eeff37767d9057992f8515106d7809ce090d3390a712f")
+    version("1.2.3", sha256="edba5d9e6a11308f82b9c8b61871e47a8ae18493bf8bff7b6ff4f4a4369428de")
     version("1.2.2", sha256="e19f49ad97bf2deedf0557eb00267f4dcf1c932c494dd07ada07fcdf5421935f")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/malt/package.py]: Cannot confirm this with the new `url`.  The sha256 for this version at that url is now `543cace664203fd9eb6b7d4945c573a3e507a43da105b5dc7ac03c78e9bb1a10`.

File: var/spack/repos/builtin/packages/malt/package.py [hunk 2]
```diff
@@ -15,12 +15,14 @@ class Malt(CMakePackage):
...
     version("1.2.1", sha256="0e4c0743561f9fcc04dc83457386167a9851fc9289765f8b4f9390384ae3618a")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/malt/package.py]: The new `url` does not support this version.  One option is to add a `url` argument:

```suggestion
    version(
        "1.2.1",
        sha256="0e4c0743561f9fcc04dc83457386167a9851fc9289765f8b4f9390384ae3618a",
        url="https://github.com/memtt/malt/archive/v1.2.1.tar.gz",
   )
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the new sha256s but not the older ones.  I made suggestions for fixing the older ones.

File: var/spack/repos/builtin/packages/malt/package.py (hunk 1 above)
Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/malt/package.py]: I fixed the checksums, using the checksum you found: I was able to build all versions now.

**Review by bernhardkaindl — APPROVED** [review]

**Review by bernhardkaindl — APPROVED** [review]
Build results on Ubuntu 22.04.4 LTS after fix of on sha265sum:
```py
bin/spack find --variants malt %clang@15.0.7 malt %clang@14.0.0 malt %gcc@9.5.0 malt %gcc@8.4.0 malt %gcc@13.1.0 malt %gcc@12.3.0 malt %gcc@11.4.0 malt %gcc@10.5.0 malt~nodejs %clang@15.0.7 malt~nodejs %clang@14.0.0 malt~nodejs %gcc@9.5.0 malt~nodejs %gcc@8.4.0 malt~nodejs %gcc@13.1.0 malt~nodejs %gcc@12.3.0 malt~nodejs %gcc@11.4.0 malt~nodejs %gcc@10.5.0
-- linux-ubuntu22.04-skylake / clang@14.0.0 ---------------------
malt@1.2.1~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.2~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.3~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.4~ipo~nodejs build_system=cmake build_type=Release generator=make
malt@1.2.4~ipo+nodejs~qt build_system=cmake build_type=Release generator=make

-- linux-ubuntu22.04-skylake / clang@15.0.7 ---------------------
malt@1.2.1~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.2~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.3~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.4~ipo~nodejs build_system=cmake build_type=Release generator=make
malt@1.2.4~ipo+nodejs~qt build_system=cmake build_type=Release generator=make

-- linux-ubuntu22.04-skylake / gcc@8.4.0 ------------------------
malt@1.2.4~ipo~nodejs build_system=cmake build_type=Release generator=make

-- linux-ubuntu22.04-skylake / gcc@9.5.0 ------------------------
malt@1.2.4~ipo~nodejs build_system=cmake build_type=Release generator=make

-- linux-ubuntu22.04-skylake / gcc@10.5.0 -----------------------
malt@1.2.1~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.2~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.3~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.4~ipo~nodejs build_system=cmake build_type=Release generator=make
malt@1.2.4~ipo+nodejs~qt build_system=cmake build_type=Release generator=make

-- linux-ubuntu22.04-skylake / gcc@11.4.0 -----------------------
malt@1.2.1~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.2~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.3~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.4~ipo~nodejs build_system=cmake build_type=Release generator=make
malt@1.2.4~ipo+nodejs~qt build_system=cmake build_type=Release generator=make

-- linux-ubuntu22.04-skylake / gcc@12.3.0 -----------------------
malt@1.2.1~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.2~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.3~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.4~ipo~nodejs build_system=cmake build_type=Release generator=make
malt@1.2.4~ipo+nodejs~qt build_system=cmake build_type=Release generator=make

-- linux-ubuntu22.04-skylake / gcc@13.1.0 -----------------------
malt@1.2.1~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.2~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.3~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
malt@1.2.4~ipo~nodejs build_system=cmake build_type=Release generator=make
malt@1.2.4~ipo+nodejs~qt build_system=cmake build_type=Release generator=make
```
Generated by https://github.com/spack/build-quality-tools/blob/main/build_pr_changes.py
```py
build-quality-tools/build_pr_changes.py -k malt -b malt,malt~nodejs -c all -ma
```

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**